packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -9769,12 +9769,13 @@ namespace Tpetra {
 #endif
         Teuchos::RCP<Teuchos::ParameterList> plist = rcp(new Teuchos::ParameterList());
         // 25 Jul 2018: Test for equality with the non-isMM path's Import object.
-        MyImport = rcp ( new import_type (MyDomainMap,
-                                          MyColMap,
-                                          RemotePids,
-                                          userExportLIDs.view(0,iloc).getConst(),
-                                          userExportPIDs.view(0,iloc).getConst(),
-                                          plist)
+        if ((MyDomainMap != MyColMap) && (!MyDomainMap->isSameAs(*MyColMap)))
```

Comment:
It's annoying that `Tpetra::Import` doesn't take care of this special case for us -- it appears in a few different places and has confused developers before.  That's future work, though :-) .